### DISCUSSION TITLE
Should tests be included in collection tarballs?

### DISCUSSION THREAD

**Discussion by gundalow** [ansible-collections/overview#111] [Proposals]
Including `tests/` in the built Ansible Galaxy package takes up a lot of space, which has to be download every time.

Are collection tarballs for users or developers?

If we say the collection tarballs are for users then:
* Add `build_ignore` in `galaxy.yml` for tests/{integration,unit,utils,output}/` (and in collection_template
* Note that we must include test/sanity (for the ignore files)
* Make it clear that contributors should use the source directly from Git.

Need to check with legal first, also package maintainers

We'd :heart: your feedback!

* Under what circumstances would this feature be useful to you?
* Is there anything you expected to see, but didn't?
* Does this have a direct negative impact on your user or developer workflow in any way?
  * ie do you develop from installed collections rather than git repo

**Comment by rndmh3ro** [comment]
Tests do not belong in the tarball. Tarballs are for end users (that's why docs do need to go into the tarball).
Developers that want to work on the collection will eventually need git and the repo of the collection to their changes upstream.

**Comment by rockaut** [comment]
I never used tests in a usage use case. And on contributions or modifications I pulled the repository anyways. So imo remove test and bring down the footprint.

**Comment by sean-m-sullivan** [comment]
Agree with the others, tests don't need to be distributed.
